### PR TITLE
work on drag drop

### PR DESCRIPTION
**Pull request by jogibear9988**

### CODE REVIEW

**Comment by jogibear9988** [comment]
any issues with this?

File: src/types.ts [hunk 1]
```diff
@@ -889,17 +889,17 @@ export type DndOptionsType = {
...
    * @default null
    */
-  dragExpand?: null | ((e: WbNodeEventType & { event: DragEvent }) => void);
+  dragExpand?: null | ((e: WbNodeEventType & { event: DragEvent }) => boolean);
   /**
    * Callback(targetNode, data)
    * @default null
    */
-  drop?:
+  dragDrop?:
```
Comment (mar10) [inline src/types.ts:902]: We should leave the name as `drop` 
If we really want change anything, then it should be 'dragDrag' to 'drag' (See https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API )
Note that changing also requires updating our docs.

File: src/wb_ext_dnd.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export class DndExtension extends WunderbaumExtension<DndOptionsType> {
       dragEnter: null, // Callback(targetNode, data), return true, to enable dnd drop
       dragOver: null, // Callback(targetNode, data)
       dragExpand: null, // Callback(targetNode, data), return false to prevent autoExpand
-      drop: null, // Callback(targetNode, data)
+      dragDrop: null, // Callback(targetNode, data)
```
Comment (mar10) [inline src/wb_ext_dnd.ts:61]: We should leave the name as drop

File: src/wb_ext_dnd.ts [hunk 3]
```diff
@@ -383,12 +386,14 @@ export class DndExtension extends WunderbaumExtension<DndOptionsType> {
...
       // Instead we remove the marker on dragenter
+      if (dndOpts.dragLeave)
+        targetNode._callEvent("dnd.dragLeave", { event: e });
       // --- drop ---
     } else if (e.type === "drop") {
       e.stopPropagation(); // prevent browser from opening links?
       this._leaveNode();
       const region = this.lastDropRegion;
-      targetNode._callEvent("dnd.drop", {
+      targetNode._callEvent("dnd.dragDrop", {
```
Comment (mar10) [inline src/wb_ext_dnd.ts:396]: We should leave the name as drop

**Review by mar10 — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 1 above)
Comment (jogibear9988) [reply src/types.ts:902]: only thought to rename it cause of dragDrag. i now rename dragDrag to only drag

**Review by mar10 — APPROVED** [review]

File: src/wb_ext_dnd.ts [hunk 4]
```diff
@@ -258,14 +258,15 @@ export class DndExtension extends WunderbaumExtension<DndOptionsType> {
 
       // --- drag ---
     } else if (e.type === "drag") {
-      // This event occurs very often...
+      if (dndOpts.drag) srcNode._callEvent("dnd.dragDrag", { event: e });
```
Comment (mar10) [inline src/wb_ext_dnd.ts]: Here we another `dragDrag` (I tried to fix it in your workOnDD branch, but it didn't appear here)

**Review by mar10 — CHANGES_REQUESTED** [review]

File: src/wb_ext_dnd.ts (hunk 4 above)
Comment (jogibear9988) [reply src/wb_ext_dnd.ts]: think it was only a github UI problem. But I rebased and force pushed now.

**Review by mar10 — APPROVED** [review]

**Comment by jogibear9988** [comment]
@mar10 could you also create a new release from the current source?

**Pull request merged**